apps/condo/domains/ticket/schema/Ticket.js

```diff
@@ -905,6 +907,56 @@ const Ticket = new GQLListSchema('Ticket', {
 
             const [requestData] = args
 
+            const { existingItem, updatedItem, operation, context } = requestData
+
+            if (operation === 'update') {
+                const ticketId = updatedItem.id
+                const isPropertyChanged = existingItem.property !== updatedItem.property
+                const isUnitChanged = existingItem.unitName !== updatedItem.unitName || existingItem.unitType !== updatedItem.unitType
+                const isClientInfoChanged = existingItem.clientName !== updatedItem.clientName || existingItem.clientPhone !== updatedItem.clientPhone
+                const isTicketCanceled = existingItem.status !== STATUS_IDS.DECLINED && updatedItem.status === STATUS_IDS.DECLINED
+
+                if (isPropertyChanged || isUnitChanged || isClientInfoChanged || isTicketCanceled) {
+                    const invoicesWithTicket = await find('Invoice', {
+                        ticket: { id: ticketId },
+                        deletedAt: null,
+                    })
```

Comment:
```suggestion
                    })
                    
                    if (invoicesWithTicket.length > 0) {
```
To prevent calculations before the possibly empty loop